src/table.js

```diff
@@ -538,15 +546,92 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type) {
+  switch (type) {
+    case "string":
+      return typeof value === "string" || value == null
+        ? value
+        : String(value);
+    case "boolean":
+      if (typeof value === "string") {
+        const trimValue = value.trim();
+        return trimValue === "true" ? true : trimValue === "false" ? false : null;
+      }
+      return typeof value === "boolean" || value == null
+        ? value
+        : Boolean(value);
+    case "bigint":
+      return typeof value === "bigint"
+        ? value
+        : value === 0 || value === true || value === false
+        ? // eslint-disable-next-line no-undef
+          BigInt(value)
+        : !value || isNaN(value) || !Number.isInteger(+value)
+        ? NaN
+        : // eslint-disable-next-line no-undef
+          BigInt(value);
+    case "integer": // not a target type for coercion, but can be inferred
+    case "number": {
+      return value === 0
+        ? value
+        : !value || isNaN(value)
+        ? NaN
+        : Number(value);
```

Comment:
I’m still a little uncertain on how we want to handle null/undefined, but how about this? In particular, this coerces `false` to `0` instead of `NaN`. Though, we might also want to prevent empty strings from getting coerced to `0` and instead coerce them to `NaN`?

```suggestion
      return typeof value === "number"
        ? value
        : value == null
        ? NaN
        : Number(value);
```

Reply:
It does seem like we might want to coerce empty strings to `NaN` rather than `0`, so they won't throw off any computations.